roles/sites-json/tasks/main.yml

```diff
@@ -8,145 +8,70 @@
-  debug:
-    var: sites_json
-  when: print_debug_messages == "TRUE"
-
-#
-# Creates a format that looks like:
-# {
-#     "id": 12345,
-#     "site": "crceducation",
-#     "url": "crceducation-dev.sites.stanford.edu"
-# },
-#
-# From something that looks like
-#  {
-#     "clayman.dev-cardinalsites.acsitefactory.com": {
-#     "name": "abc123dev",
-#     "flags": [],
-#     "conf": {
-#       "gardens_site_id": 12345,
-#       "gardens_db_name": "asdfasdfasdfasdf",
-#       "acsf_site_id": 12345,
-#       "acsf_db_name": "abc123dev",
-#       "site_api_key": "456789235489789654897984135897564"
-#     }
-#   },
-#
-# The REST API has a restriction of 100 items per request and quickly falls over
-# for our needs. To bypass the API we are directly getting all of the site info
-# from sites.json on the acquia server. Unfortunately the initial format is not
-# usable and we need to parse out the machine name and id of the site factory
-# site by using the url for the sitefactory item as the machine name and getting
-# the key from the json array for the node id. We only use the acsitefactory uri
-# as it is the canonical machine name for the site and the other domains listed
-# cannot be trusted.
-#
-# In this task, we use a loop to build a dict. On the first pass, if it does not
-# exist we instantiate it with an empty array |default([]), then append the new
-# array of values to it. On the subsequent passes it gets more values appended.
-#
-- name: Manipulate sites.json
-  set_fact:
-    sites_formatted: "{{ sites_formatted|default([]) + [ {'url': item.0, 'id': sites_json.sites[item.0].conf.acsf_site_id, 'site': item.0|regex_replace('.(dev-|test-|)cardinalsites.acsitefactory.com', '') } ] }}"
-  with_together:
-    - "{{ sites_json.sites }}"
-    - "{{ sites_json.sites }}"
-  when: item.0 | regex_search('acsitefactory')
+# There are several entries for each site within the sites.json file.
+# Fortunately, each entry is on its own line. We know that
+# "sitename.cardinalsites.acsitefactory.com" is the canonical name for a given
+# site. Grep for that and output it to a file.
+- name: Grep sites.json file for our sitename
+  shell: "grep {{ acsf_site_name }}.{{ sitefactory_environment }}cardinalsites.acsitefactory.com /tmp/{{ inventory_hostname }}/sites.json > /tmp/{{ inventory_hostname }}/sites_json_raw"
+  ignore_errors: true
 
-# Debug the sites_formatted array.
-#
-# The array will be output in alphabetical order (id, site, url), but as the keys
-# are object properties, the order does not matter.
-#
-- name: debug sites_formatted
-  debug:
-    var: sites_formatted
-  when: print_debug_messages == "TRUE"
+# sites_json_raw will be a really long string like this:
+# "jbickar.cardinalsites.acsitefactory.com": {"name":"sbiybj3061","flags":{"preferred_domain":true},"conf":{"gardens_site_id":3061,"gardens_db_name":"sbiybj3061","acsf_site_id":3061,"acsf_db_name":"sbiybj3061","site_api_key":"fc28cad31e56b5ae1e97629348872878276510bc"}},
```

Comment:
Probably don't want to publish these ids and keys.

Reply:
Well, nuts.